src/text-editor-presenter.coffee

```diff
@@ -1236,7 +1254,12 @@ class TextEditorPresenter
     range = marker.getScreenRange()
 
     if decoration.isDestroyed() or not marker.isValid() or range.isEmpty() or not range.intersectsRowRange(@startRow, @endRow - 1)
-      delete @state.content.highlights[decoration.id]
+      tileStartRow = @tileForRow(range.start.row)
```

Comment:
This code is repeated a couple of times now. I am thinking we could extract something like:

```coffee
iterateTiles: (startRow, endRow, fn)
```

Which could basically perform the conversion from row to tile along with the necessary setup (e.g. initialize the tile state).